lib/logstash/inputs/s3.rb

```diff
@@ -146,7 +151,11 @@ def list_new_files
     rescue Aws::Errors::ServiceError => e
       @logger.error("S3 input: Unable to list objects in bucket", :prefix => prefix, :message => e.message)
     end
-    objects.keys.sort {|a,b| objects[a] <=> objects[b]}
+    if @sort_processed_files
+      objects.keys.sort {|a,b| objects[a] <=> objects[b]}
+    else
+      objects.keys.shuffle {|a,b| objects[a] <=> objects[b]}
```

Comment:
@anas-aso I don't know the API being used here, but typically in ruby methods that mutate objects have a `!` at the end of them. I would expect these functions to return a sorted/shuffled list without mutation.